packages/components/src/components/assay/AssayImportPanels.tsx

```diff
@@ -370,8 +383,11 @@ class AssayImportPanelsImpl extends React.Component<Props, State> {
     };
 
     handleDataTextChange = (inputName: string, fieldValues: any) => {
+        if (this.props.onDataChange) {
+            this.props.onDataChange(fieldValues !== undefined && fieldValues !== '', IMPORT_DATA_FORM_TYPES.TEXT);
```

Comment:
Same comment as above. Maybe we should just always have these handler return dirty true and then the cancel will put that back to false or the successful import/submit. What do you think?